lookout/style/format/annotations/annotated_data.py

```diff
@@ -0,0 +1,329 @@
+                    right_span, right_annotation = \
+                        self._type_to_annotations[annotation_type].peekitem(right_span_index)
+                    if self._check_spans_overlap(*right_span, *annotation.span):
+                        raise ValueError(
+                            "Trying to add Annotation %s which is overlaps with existing %s" % (
+                                right_annotation, annotation))
+                left_span_index = max(0, right_span_index - 1)
+                left_span, left_annotation = self._type_to_annotations[annotation_type].peekitem(
+                    left_span_index)
+                if self._check_spans_overlap(*left_span, *annotation.span):
+                    raise ValueError("Trying to add Annotation %s which is overlaps with existing"
+                                     "%s" % (left_annotation, annotation))
+            self._span_to_annotations[annotation.span][annotation_type] = annotation
+            self._type_to_annotations[annotation_type][annotation.span] = annotation
+
+    def get(self, annotation_type: Type[Annotation], span: Optional[Tuple[int, int]] = None,
+            ) -> Annotation:
+        """
+        Return an annotation for the given span and type.
+
+        Looking for an exact match only.
+
+        :param annotation_type: Annotation type to get.
+        :param span: Annotation span (range) to get. If span is not specified it returns an \
+                     annotation that cover all content (aka global annotation).
+        :return: Requested `Annotation`.
+        """
+        if span is None:
+            return self._global_annotations[annotation_type]
+        else:
+            check_span(*span)
+            return self._type_to_annotations[annotation_type][span]
+
+    def iter_annotations(self, annotation_type: Type[Annotation],
+                         *additional: Type[Annotation],
+                         start_offset: Optional[int] = None) -> Union[Iterator[AnnotationsSpan]]:
+        """
+        Iterate through annotations with specified type.
+
+        Iteration goes through `annotation_type`. It is additionally annotated with annotations
+        specified in `additional`. `additional` can't be empty. If you need to iterate through \
+        one annotation only use `AnnotationManager.iter_annotation()`.
+
+        :param annotation_type: Type of annotation to iterate through.
+        :param additional: Additional annotations that should be added to the main one.
+        :param start_offset: Start to iterate from the spesific offset. \
+                             Can be used as a key argument only.
+        :return: Iterator through annotations of requested types.
+        """
+        if not additional:
+            raise ValueError("At least one additional annotation should be specified. "
+                             "If you need to iterate through only one annotation use "
```

Comment:
Ok, I add a good explanation about the difference between `iter_annotation` and `iter_annotations` to a docstring. They are similar indeed, but since they return different types I do not want to merge these functions. PTAL and tell me if it is clear now. 

Question how to name `iter_annotations` remains open for now.